### ISSUE TITLE
Section: GDB

### ISSUE THREAD

**Issue by ryankurte** [rust-embedded/debugonomicon#11]
An introduction to our old friend GDB

- [ ] How it works (gdb server vs. client etc.)
- [ ] How to run against a compiled rust target
- [ ] Common debugging commands
- [ ] Useful references

**Comment by frroossst** [comment]
I was wondering if help was still needed to complete this, or if it's a dead project

**Comment by ryankurte** [comment]
howdy! i still think it'd be a super helpful resource if you're interested in contributing ^_^

**Comment by frroossst** [comment]
Yes, I would love to contribute (bear in mind, I'm still a beginner on the embedded Rust side of things). I just wanted to help make embedded Rust more accessible to more people. 

Any tips on writing this section? How detailed should this be? How abstracted or lower level etc?

**Comment by eldruin** [comment]
I would say let's start with something high-level in line with the TODO items posted above. 
You can start by filling in the sections [here](https://github.com/rust-embedded/debugonomicon/blob/master/src/introduction.md) but please feel free to rename, rearrange and add sections as you see fit.
Once you send a PR we can discuss in detail.
